pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -521,7 +523,7 @@ func (p *PriorityQueue) flushBackoffQCompleted() {
 			klog.ErrorS(err, "Unable to pop pod from backoff queue despite backoff completion", "pod", klog.KObj(pod))
 			break
 		}
-		if added, _ := p.addToActiveQ(pInfo); added {
+		if added, _ := p.addToActiveQ(pInfo, true); added {
```

Comment:
This is the only case that we skip preEnqueue checks, so it makes more sense to simply make the change here, without changing the signature of `addToActiveQ`:

```go
if err := p.activeQ.Add(pInfo); err == nil {
```

Reply:
and could you add a test to cover the change? (ensure PreEnqueue is **_not_** called during flushBackoffQCompleted().